front/src/monthview/Monthview.js

```diff
@@ -0,0 +1,479 @@
+    const yyyy = today.getFullYear();
+    this.setState({ yearNro: yyyy });
+    return yyyy;
+  }
+
+  getMonth = () => {
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramMonth = urlParamDateSplit[1].padStart(2, '0');
+      if (paramMonth.length === 2) {
+        this.setState({ monthNro: paramMonth });
+        return paramMonth;
+      }
+    } catch {
+      console.log('Error getting month from url');
+    }
+    const today = new Date();
+    const mm = String(today.getMonth() + 1).padStart(2, '0');
+    this.setState({ monthNro: mm });
+    return mm;
+  }
+
+  createWeekDay = () => {
+    const table = [];
+    let dayNumber;
+
+    const fin = localStorage.getItem('language');
+    const { month } = texts;
+
+    for (let j = 0; j < 8; j += 1) {
+      dayNumber = j.toString();
+
+      if (dayNumber === '0') {
+        table.push(
+          <div className="weekNumber">
+            {month.weekNumber[fin]}
+          </div>,
+        );
+      } else {
+        table.push(
+          <div>
+            {weekdayShorthand[dayNumber - 1][fin]}
+          </div>,
+        );
+      }
+    }
+
+    return table;
+  }
+
+  createMonthTable = () => {
+    const requestSchedulingFreeform = async (date) => {
+      const response = await getSchedulingFreeform(date);
+
+      if (response) {
+        this.setState({
+          state: 'ready',
+          paivat: response.month,
+        });
+      } else console.error('getting info failed');
+    };
+
+    const table = [];
+    if (this.state.yearNro === 0) {
+      return false;
+    }
+
+    let help = 0;
+    const days = moment(`${this.state.yearNro}-${this.state.monthNro}`, 'YYYY-MM').daysInMonth();
+    const startDay = moment(`${this.state.yearNro}-${this.state.monthNro}-1`, 'YYYY-MM-DD');
+    let firstMon = moment(`${this.state.yearNro}-${this.state.monthNro}-1`, 'YYYY-MM-DD');
+    let info = false;
+
+    // safety loop incase of errors in name detection due to wrong locale, e.g. Sweden
+    let safetyLoop = 0;
+
+    while (firstMon.format('ddd') !== 'Mon' && firstMon.format('ddd') !== 'ma') {
+      firstMon = firstMon.subtract(1, 'days');
+      if (safetyLoop > 7) {
+        break;
+      }
+      safetyLoop += 1;
+    }
+    safetyLoop = 0;
+
+    if (this.state.paivat === undefined || this.state.paivat[0].date !== firstMon.format('YYYY-MM-DD')) {
+      requestSchedulingFreeform(firstMon.format('YYYY-MM-DD'));
+    }
+    if (this.state.paivat === undefined) {
+      return false;
+    }
+    // Tarkistetaan ensimmäinen maanantai ja lisätään siitä päivät
+    while (firstMon.format('ddd') !== startDay.format('ddd')) {
+      const colorFromBackEnd = checkColor(this.state.paivat, help);
+      if (this.state.paivat[help].tracks) {
+        this.state.paivat[help].tracks.forEach((track) => {
+          if (track.notice !== null) {
```

Comment:
Should here be the same empty string check?

Reply:
Added empty string check